src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -1061,17 +1272,117 @@ def is_finite(self):
         from sage.rings.function_field.drinfeld_modules.finite_drinfeld_module import FiniteDrinfeldModule
         return isinstance(self, FiniteDrinfeldModule)
 
-    def j_invariant(self):
+    def is_isomorphic(self, psi):
         r"""
-        Return the j-invariant of the Drinfeld module if the rank is
-        two; raise a NotImplementedError otherwise.
+        Return ``True``  whether ``self`` is isomorphic to the Drinfeld module
+        `\psi`.
 
-        Assume the rank is two. Write the generator `\phi_T = \omega +
-        g\tau + \Delta\tau^2`. The j-invariant is defined by
-        `\frac{g^{q+1}}{\Delta}`, `q` being the order of the base field
-        of the function ring. In our case, this field is always finite.
+        EXAMPLES::
 
-        OUTPUT: an element in the base codomain
+            sage: A = GF(5)['T']
+            sage: K.<T> = Frac(A)
+            sage: c = T^4 + 3*T^2 + T
+            sage: phi = DrinfeldModule(A, [T, T^3, T^9, T])
+            sage: psi = DrinfeldModule(A, [T, c^4*T^3, c^(24)*T^9, c^(124)*T])
+            sage: phi.is_isomorphic(psi)
+            True
+            sage: phi0 = DrinfeldModule(A, [T, 1, 1, 1])
+            sage: phi.is_isomorphic(phi0)
+            False
+
+        ::
+
+            sage: phi = DrinfeldModule(A, [T, 1, 0, 1, 0, 0, 0, T^2, 1, 1])
+            sage: phi.is_isomorphic(phi)
+            True
+            sage: psi = DrinfeldModule(A, [T, 1, 0, 1, 0, T, 0, T^2, 1, 1])
+            sage: phi.is_isomorphic(psi)
+            False
+        """
+        # Trivial checks:
+        if self._gen == psi._gen:
+            return True
+        if self._gen.degree() != psi._gen.degree():
+            return False
+        if self._gen.degree() == 1:
+            return True
+        # Check if self and psi are patterned alike:
+        if not all(bool(g_self) == bool(g_psi) for g_self, g_psi in\
+            zip(self._gen.coefficients(sparse=False),
+                psi._gen.coefficients(sparse=False))):
+            return False
+        # Check that all the nonzero basic j-invariants agree
+        isom = True
+        for p in self.basic_j_invariant_parameters(nonzero=True):
+            if self.j_invariant(p, check=False) != psi.j_invariant(p,check=False):
+                isom = False
+                break
+        return isom
+
+    def j_invariant(self, parameter=None, check=True):
+        r"""
+        Return the `j`-invariant of the Drinfeld
+        `\mathbb{F}_q[T]`-module for the given parameter.
```

Comment:
In the rest of the codebase, we do not specify $\mathbb{F}_q[T]$. However, you can specify in the docstring that the mathematical construction is only for Drinfeld modules over $\mathbb{F}_q[T]$.

```suggestion
        module for the given parameter.
```

Reply:
I'd like to keep that for three reasons:

1. Yes, in the rest of the code base we do in fact say "Drinfeld $\mathbb{F}_q[T]$-module" (check back your code).
2. "Drinfeld module over $\mathbb{F}_q[T]$" is slightly incorrect and an abuse of notation (or definition?) since this means that $\mathbb{F}_q[T]$ is the base ring of the Drinfeld module which is not true (its base ring is a $A$-field $F$, usually a (not necessarily finite) field extention of $\mathrm{Frac}(A)$ or a finite field).
3. In the events were we generalize for general $A$, then specifying "Drinfeld $A$-module" will be mandatory

Reply:
While it is true that we sometimes write the prefix `\mathbb{F}_q[T]`, in the context of first docstring paragraphs, we mostly just say `the Drinfeld module` (see e.g. `action`), and this is what I was referring to. But I'm fine with keeping it!